tests/tensor/test_nlinalg.py

```diff
@@ -580,3 +581,41 @@ def test_eval(self):
         t_binv1 = tf_b1(self.b1)
         assert _allclose(t_binv, n_binv)
         assert _allclose(t_binv1, n_binv1)
+
+
+class TestKron(utt.InferShapeTester):
+    rng = np.random.default_rng(43)
+
+    def setup_method(self):
+        self.op = kron
+        super().setup_method()
+
+    def test_perform(self):
+        for shp0 in [(2,), (2, 3), (2, 3, 4), (2, 3, 4, 5)]:
```

Comment:
Same here? Use pytest.mark.parametrize?

Reply:
hey! could you briefly explain how `pytest.mark.parametrize`  works. It'll anyways be useful in writing future tests and I can update these as well

Reply:
It's a decorator that passes keyword arguments into a test function. For example:

```
@pytest.mark.parameterize('a', [1,2,3])
def test(a):
    assert a < 2
````

This will run 3 tests, one for each value of `a`. The important thing is that the arguments into the test function match the strings you give in the decorator. 

You can pass multiple parameters like this:

```
@pytest.mark.parameterize('a, b', [(1, 2) ,(2, 3), (3, 4)])
def test(a, b):
    assert a < b
```
This will make 3 tests, all of which will pass.

Or you can make the full cartesian product between parameters by stacking decorators:

```
@pytest.mark.parameterize('a', [1, 2, 3])
@pytest.mark.parameterize('b', [3, 4, 5])
def test(a, b):
    assert a < b
```

This will make `3 * 3 = 9` tests, and you should get one failure (for the `3 < 3` case)

Reply:
thank you this is really helpful. safe to say these are just fancy for loops? :smile:

Reply:
Yes, but with better outputs when you run pytest. It splits each parameterization into its own test, and so you can know exactly which combination of parameters is failing. If you have a loop, it just tells you pass/fail, not at which step of the loop.